plugins/inputs/sql/sql.go

```diff
@@ -357,10 +368,54 @@ func (s *SQL) Start(_ telegraf.Accumulator) error {
 	// Connect to the database server
 	s.Log.Debugf("Connecting to %q...", s.Dsn)
 	s.db, err = dbsql.Open(s.driverName, s.Dsn)
+	if err != nil {
+		if !s.AllowConnectionFailures {
+			return err
+		}
+		// if connection failures allowed, periodically attempt to reconnect to the DB
+		go s.connect()
```

Comment:
Should also return here to not start below (start is called from within connect)